src/snapred/backend/service/CalibrationService.py

```diff
@@ -196,29 +197,82 @@ def fitPeaks(self, request: FitMultiplePeaksRequest):
 
     @FromString
     def save(self, request: CalibrationExportRequest):
+        """
+        If no version is attached to the request, this will save at next version number
+        """
         entry = request.calibrationIndexEntry
-        version = entry.version
-        calibrationRecord = request.calibrationRecord
-        if version is not None:
-            calibrationRecord.version = version
-        calibrationRecord = self.dataExportService.exportCalibrationRecord(calibrationRecord)
-        calibrationRecord = self.dataExportService.exportCalibrationWorkspaces(calibrationRecord)
-        entry.version = calibrationRecord.version
-        self.saveCalibrationToIndex(entry)
+        record = request.calibrationRecord
+        version = self.dataFactoryService.getThisOrNextCalibrationVersion(
+            record.runNumber,
+            record.useLiteMode,
+            request.version,
+        )
+
+        # Rebuild the workspace names to strip any "iteration" number:
+        savedWorkspaces = {}
+        for key, wsName in record.workspaces.items():
+            match key:
+                case wngt.DIFFCAL_OUTPUT:
+                    if wng.Units.DSP.lower() in wsName:
+                        savedWorkspaces[key] = (
+                            wng.diffCalOutput()
+                            .unit(wng.Units.DSP)
+                            .runNumber(record.runNumber)
+                            .version(version)
+                            .group(record.focusGroupCalibrationMetrics.focusGroupName)
+                            .build()
+                        )
+                    else:
+                        raise RuntimeError(
+                            f"""
+                            cannot save a workspace-type: {wngt.DIFFCAL_OUTPUT}
+                            without a units token in its name {wsName}
+                            """
+                        )
+                case wngt.DIFFCAL_DIAG:
+                    savedWorkspaces[key] = (
+                        wng.diffCalOutput()
+                        .unit(wng.Units.DIAG)
+                        .runNumber(record.runNumber)
+                        .version(version)
+                        .group(record.focusGroupCalibrationMetrics.focusGroupName)
+                        .build()
+                    )
+                case wngt.DIFFCAL_MASK:
+                    savedWorkspaces[key] = wng.diffCalMask().runNumber(record.runNumber).version(version).build()
+                case wngt.DIFFCAL_TABLE:
+                    savedWorkspaces[key] = wng.diffCalTable().runNumber(record.runNumber).version(version).build()
+                case _:
+                    raise RuntimeError(f"Unexpected output type {key} for {wsName}")
+
+            self.groceryService.renameWorkspace(wsName, savedWorkspaces[key])
+
+        record.workspaces = savedWorkspaces
```

Comment:
This was previously taking place inside the LocalDataService, and required being called twice with nearly duplicated code.  Moving it here makes more sense, and also prevents the duplicate renaming code.